### PR TITLE
[EXPORTER] Change OTLP HTTP content_type default to binary

### PR DESCRIPTION
**Pull request by kkuehl**
https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md

Fixes # (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by ThomsonTan** [comment]
@kkuehl thanks for the contribution. Could you please sign the CLA?

File: exporters/otlp/test/otlp_http_exporter_test.cc [hunk 1]
```diff
@@ -106,7 +106,7 @@ class OtlpHttpExporterTestPeer : public ::testing::Test
   void ExportJsonIntegrationTest()
   {
     auto mock_otlp_client =
-        OtlpHttpExporterTestPeer::GetMockOtlpHttpClient(HttpRequestContentType::kJson);
+        OtlpHttpExporterTestPeer::GetMockOtlpHttpClient(HttpRequestContentType::kBinary);
```
Comment (lalitb) [inline exporters/otlp/test/otlp_http_exporter_test.cc:109]: Can you check this change again, as the function name indicates this doing http/json test?

**Comment by lalitb** [comment]
Thanks for working on it. Also, please add the changelog as this is supposed to be a breaking change.

File: exporters/otlp/test/otlp_http_exporter_test.cc (hunk 1 above)
Comment (kkuehl) [reply exporters/otlp/test/otlp_http_exporter_test.cc:109]: pushed, thanks.

**Comment by kkuehl** [comment]
> @kkuehl thanks for the contribution. Could you please sign the CLA?

I have signed the CA, but I assume there is a manual step.

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ Breaking changes:
   * CMake options `WITH_OTLP_HTTP_SSL_PREVIEW`
     and `WITH_OTLP_HTTP_SSL_TLS_PREVIEW` are removed.
     Building opentelemetry-cpp without SSL support is no longer possible.
-
+* [#240] Change default OTLP Http Exporter content type from kJson to kBInary
```
Comment (marcalff) [inline CHANGELOG.md:36]: The PR number is 2480

What is the rationale for this change ?

**Review by marcalff — COMMENTED** [review]
Thanks for the patch.

The spec does not say (unless I missed it) to use binary of json by default,
what is the motivation for this change ?

File: CHANGELOG.md (hunk 2 above)
Comment (marcalff) [reply CHANGELOG.md:36]: Found it, the proper link is:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specify-protocol

The default is http/protobuf, which means binary.

Comment (kkuehl) [reply CHANGELOG.md]: > The PR number is 2480
> 
> What is the rationale for this change ?

Fixed PR number.

Comment (kkuehl) [reply CHANGELOG.md]: This change is also a 3x to 10x performance improvement in my tests.

File: exporters/otlp/test/otlp_http_exporter_test.cc (hunk 1 above)
Comment (lalitb) [reply exporters/otlp/test/otlp_http_exporter_test.cc:109]: I meant this function does the JSON test, it can remain unchanged.

**Comment by marcalff** [comment]
@kkuehl 

About the CLA, somehow the system sees two accounts : @kkuehl and @kirby-mimic 

Please make sure you sign the CLA for the proper account.

This needs to be resolved, as we can not merge a PR without it.

Start over and create another clean PR if you have to.
